scripts/helpers/check_zap_against_xml.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+import click
+
+DATA_MODEL_PATH = 'data_model/clusters'
+
+
+@click.command()
+@click.option(
+    '--zap',
+    required=True,
+    type=str,
+    help='ZAP file')
+@click.option(
+    '--matter-root',
+    required=True,
+    type=str,
+    help='Path to the matter root')
+def main(zap, matter_root):
+
+    all_checks_successful = True
+
+    CLUSTER_REVISIONS_BY_CLUSTER_ID = {}
+
+    print('Getting latest cluster revisions from XML data model')
+    for file in os.listdir(os.path.join(matter_root, DATA_MODEL_PATH)):
+        if file.endswith(".xml"):
+            with open(os.path.join(matter_root, DATA_MODEL_PATH, file), "r") as cluster_xml_file:
+                print(f'Processing {file}')
+                xml_et = ET.parse(cluster_xml_file)
+                root = xml_et.getroot()
+
+                try:
+                    CLUSTER_REVISIONS_BY_CLUSTER_ID[int(root.attrib['id'], 16)] = root.attrib['revision']
+                except ValueError:
+                    print(' ⚠️ Failed to parse Cluster ID')
+
+    # print(CLUSTER_REVISIONS_BY_CLUSTER_ID)
+    print()
```

Comment:
consider using logging instead of prints.

Reply:
done.